samples/deployment/aws/build_ami_vftd.yaml

```
@@ -0,0 +1,196 @@
+        wait: yes
+        assign_public_ip: yes
+        instance_initiated_shutdown_behavior: terminate
+      register: stage_vm
+    - name: "Add the newly created host so that we can further contact it"
+      add_host:
+        name: "{{ item.public_ip }}"
+        groups: "stage_servers"
+      with_items: "{{ stage_vm.instances }}"
+    - name: "Wait for SSH to come up"
+      delegate_to: "{{ item.public_dns_name }}"
+      wait_for_connection:
+        delay: 60
+        timeout: 320
+      with_items: "{{ stage_vm.instances }}"
+
+    # Build vFTD Volume Section
+    - name: "Prepare mnt directory"
+      shell: "chmod 777 /mnt"
+      become: yes
+      delegate_to: "{{ item.public_dns_name }}"
+      with_items: "{{ stage_vm.instances }}"
+    - name: "Copy QCOW2 file to the remote VM"
+      copy:
+        src: "./Cisco_Firepower_Threat_Defense_Virtual-{{ ftd_build_id }}.qcow2"
+        dest: "/mnt/image.qcow2"
+      delegate_to: "{{ item.public_dns_name }}"
+      with_items: "{{ stage_vm.instances }}"
+    - name: "Copy fixup.sh file to the remote VM"
+      copy:
+        src: "./fixup-ftd.sh"
+        dest: "/mnt/fixup.sh"
+        mode: "u+rwx"
+      delegate_to: "{{ item.public_dns_name }}"
+      with_items: "{{ stage_vm.instances }}"
+    - name: "Run fixup.sh preinstall command"
+      shell: "/mnt/fixup.sh preinstall"
+      become: yes
+      delegate_to: "{{ item.public_dns_name }}"
+      with_items: "{{ stage_vm.instances }}"
+    - name: "Run fixup.sh doinstall command"
+      shell: "/mnt/fixup.sh doinstall"
+      become: yes
+      delegate_to: "{{ item.public_dns_name }}"
+      with_items: "{{ stage_vm.instances }}"
+    - name: "Run fixup.sh postinstall command"
+      shell: "/mnt/fixup.sh postinstall"
+      become: yes
+      delegate_to: "{{ item.public_dns_name }}"
+      with_items: "{{ stage_vm.instances }}"
+
+    # Create Custom AMI Section
+    - name: "Get volume ID"
+      ec2_instance_facts:
+        instance_ids:
+          - "{{ stage_vm.instance_ids.0 }}"
+      register: stage_vm_facts
+    - name: "Create a Snapshot"
+      ec2_snapshot:
+        instance_id: "{{ stage_vm.instance_ids.0 }}"
+        device_name: "/dev/sdc"
+        description: "Snapshot for vFTD AMI image"
+      register: stage_snapshot
+    - name: "Create Snapshot based AMI Image"
+      ec2_ami:
+        name: "vftd_{{ ftd_build_id }}.ami"
+        description: "Snapshot based AMI image for vFTD {{ ftd_build_id }}"
+        sriov_net_support: "simple"
+        architecture: "x86_64"
+        root_device_name: "/dev/xvda"
+        virtualization_type: "hvm"
+        state: present
+        device_mapping:
+          - device_name: "/dev/xvda"
+            snapshot_id: "{{ stage_snapshot.snapshot_id }}"
+
+    # Cleanup ENV
+    - name: "Destroy staging VM"
+      ec2:
+        state: 'absent'
+        instance_ids: '{{ stage_vm.instance_ids }}'
+    - name: "Destroy AWS EC2 private key"
```

```suggestion
    - name: "Destroy an EC2 key pair"
```
We are destroying the whole key pair, not private key in particular.

---

yes. fixed